### DISCUSSION TITLE
Coloban is running on Quasar 2 now, yay!

### DISCUSSION THREAD

**Discussion by pikil** [quasarframework/quasar#10462] [Show and Tell]
## Coloban 0.9 with Vue 3 and Quasar 2

Wow, that wasn't a walk in the park, but wasn't a rocket science either! :)

### :guitar: We've finally moved Coloban to V3Q2 stack!!!   :guitar:

As we have had many things to re-design and re-think over the development time, we'd decided to actually re-build parts of the project from scratch. Yes, not always a great idea, but that allowed us to implement Vue 3, Quasar 2, Capacitor 3, Electron 13 support and gave an additional space for improvements and growth.

That was a really nice adventure and the project does feel faster/neater/friendlier now.

_In short, Coloban project is a Chat application with implemented tasks management tool, which allows to communicate with teammates and create tasks in the same app with no need to switch between apps. Any message (even the past one) can be converted to a task, so there is no need to create a long ticket once again, when it is not needed._

It is a wonderful showcase of what a modern Javascript project can do. And it also shows the possibilities to build a multi-platform app with a really small team and a single code-base :) Neat!

Thank you awesome Quasar team and personally Razvan for an incredible work!!!

### The application is running on:  
- Vue 3.x;
- Quasar 2.x;
- Capacitor 3.x (Android & iOS);
- Electron 13.x;
- Web & PWA;

### Quasar extensions:  
- quasar-ui-qcalendar;
- quasar-ui-markdown;

### Local data management:
- Indexed DB and Local storage;

### Platforms
- Web;
- PWA;
- MacOS;
- Windows;
- Android;
- iOS;

### Additional information:
- Dark/Light/Auto themes;
- Burnup charts - using chart.js;
- Gantt charts - using dxhtml;
- HTTP requests - axios;
- Websockets - native JS implementation;
- Audio/Video calls and screenshare - native WebRTC serverless implementation;
- Push notifications on iOS and Android - Firebase FCM (capacitor-push-notifications);
- MacOS signing and notarization - (electron-notarize, electron-builder, electron;
- Auto-updater for the app (electron-updater);
- Windows & Macos notifications (+ badges);
- HTTPS for running WebRTC and Websockets securely;

### Available chat features:  
- Creating private and group topics within projects;
- Call/Screenshare to topic participants;
- Send chat texts, files, emojis, gifs, stickers;
- Convert messages to tasks in one click;

### Available project management  
- Create specific projects;
- Create topics within projects;
- Create tasks within topics;
- Set assignee(s) and start/due times for tasks;

### Tasks can be managed in:  
- Eisenhower;
- Calendar;
- Burnup charts with project estimation completions;
- Gantt charts (Start&Due times are obligatory for those);
- Kanban;
- Backlog;
- Directly in topic's chat;

I've added a few screenshots for reference, but you can easily jump into the project at www.coloban.com/download

Any questions - do not hesitate to leave a message here :)

Cheers, ilia!

<img width="1263" alt="Screen Shot 2021-08-24 at 10 52 12 PM" src="https://user-images.githubusercontent.com/26779040/130605841-8544c356-29fc-40ef-981b-11da8d1c0310.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 49 55 PM" src="https://user-images.githubusercontent.com/26779040/130605890-0d4b8ca3-2c71-4438-8a5f-3d671688791d.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 49 23 PM" src="https://user-images.githubusercontent.com/26779040/130605902-12b2888a-421f-4e29-be57-829da2a95011.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 48 20 PM" src="https://user-images.githubusercontent.com/26779040/130605930-e6eb7d32-25a1-442f-b709-4d06921275eb.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 52 36 PM" src="https://user-images.githubusercontent.com/26779040/130605951-9adc5a98-9448-460e-8d1a-64ccaec41dcc.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 53 59 PM" src="https://user-images.githubusercontent.com/26779040/130606000-9db757e7-273d-477b-b77a-586b025ab84b.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 57 08 PM" src="https://user-images.githubusercontent.com/26779040/130606017-243c747b-f1fe-4a56-9b2f-adf6c44d966f.png">

<img width="1263" alt="Screen Shot 2021-08-24 at 10 56 40 PM" src="https://user-images.githubusercontent.com/26779040/130606042-89d76d38-cbe0-4e43-977f-9d7ab762eb63.png">

**Comment by rstoenescu** [comment]
Congrats and keep up the good work!

**Reply by pikil** [reply]
Yeah, thanks for that!

**Comment by hawkeye64** [comment]
This is an amazing piece of software! Congrats and keep on truckin'!

**Reply by pikil** [reply]
Thank you!

**Comment by IlCallo** [comment]
I almost want to use it myself now 👀

**Reply by pikil** [reply]
You are more than welcome to do so :)

**Comment by ldiebold** [comment]
This is awesome! We need to get you on a podcast 😉

**Reply by pikil** [reply]
Absolutely! That would be really nice! :)

**Comment by webnoob** [comment]
Amazing work! Looks great.

**Comment by etewiah** [comment]
Hey, I tried it out but at this step I got a 404 :(

https://app.coloban.com/accept-invite/my-invite-code-here
